File: NapakalakiRuby/lib/monster.rb
# By: Miguel Robles Urquiza
#     Jesús Sánchez de Lechina Tejada

module NapakalakiGame
  class Monster

    attr_accessor :name, :combatLevel, :badconsequence, :prize

    def initialize(name, combatLevel,badconsequence,prize)
      @name = name
      @combatLevel = combatLevel
      @badconsequence = badconsequence
      @prize = prize
    end

    def getTreasuresGained          # Número de treasures que se obtienen al ganar
      @prize.treasures.size()
    end

    def getLevelsGained             # Levels que se obtienen al ganar
      @prize.levels
    end

    def getKill               # Indica si el monstruo tiene la capacidad de matar
      @badconsequence.death
    end

    def to_s
      "Name: #{@name} \nCombatLevel: #{@combatLevel} \nPrize: #{@prize} \nBadConsequence: #{@badconsequence} \n\n"
    end

  end
end

File: Napakalaki/src/NapakalakiGame/Monster.java
package NapakalakiGame;

/**
 *
 * @author Jesús Sánchez de Lechina Tejada & Miguel Ángel Robles Urquiza
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badconsequence;
    
    public Monster(String name, int level, Prize prize, BadConsequence bc){
        this.name = name;
        this.combatLevel = level;
        this.prize = prize;
        this.badconsequence = bc;
    }
    
    public String getString(){ return name; }    
    public int getCombatLevel(){ return combatLevel; }    
    public Prize getPrize(){ return prize; }    
    public BadConsequence getBadConsequence(){ return badconsequence; }   
    
    @Override
    public String toString(){
        return "Monster: " + name + " \nCombat level: " + Integer.toString(combatLevel) + " \nPrize: " + prize.toString() + "\nBad Consequence: " + badconsequence.toString() + "\n";
    }
}
    
